### PR TITLE
fix: plugin disposal, race, and some very rare edge case bugs

### PR DESCRIPTION
**Pull request by materwelonDhruv**
## Checklist

- [x] Tests cover the change
- [x] `pnpm prePush` is green
- [x] Changeset added with `pnpm cs`, for any change to a published package

## Summary by CodeRabbit

* **New Features**
  * Added configurable shutdown deadlines, defaulting to 25 seconds, with validation for invalid values.
  * Added shutdown drain reporting, including warnings for handlers still running when the drain window closes.
  * Late-finishing plugin initialization now triggers cleanup when appropriate.

* **Bug Fixes**
  * Shutdown continues gracefully when tunnels or in-flight handlers fail to stop.
  * Late plugin initialization failures are logged without preventing shutdown.

* **Documentation**
  * Expanded lifecycle and plugin documentation with timing, cleanup, shutdown, and configuration details.

### CODE REVIEW

File: packages/core/tests/node/pluggable-init-timeout.test.ts [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+    public override dispose(): Promise<void> {
+        this.disposeCalls++;
+        this.claimHeld = false;
+        return Promise.resolve();
+    }
+}
+
+class TestHost extends Pluggable<'gateway', 'server'> {
+    public readonly config = {} as Config;
```
[the remark this answers is not in the document]
Comment (materwelonDhruv) [reply packages/core/tests/node/pluggable-init-timeout.test.ts:42]: nah.
